graphql-ballerina/Module.md

```diff
@@ -0,0 +1,685 @@
+            "locations": [
+                {
+                    "line": 1,
+                    "column": 1
+                }
+            ]
+        }
+    ]
+}
+```
+
+### Types
+The Ballerina GraphQL resources can return the following types:
+
+#### Return Types
+
+##### Scalar types
+The following Ballerina types are considered as Scalar types:
+- `int`
+- `string`
+- `boolean`
+- `float`
+- `decimal`
+- `enum`
+
+```ballerina
+resource function get greeting() returns string {
+    return "Hello, World";
+}
+```
+
+This can be queried using the following document:
+```
+{
+    greeting
+}
+```
+
+Result:
+```json
+{
+    "data": {
+        "greeting": "Hello, World"
+    }
+}
+```
+
+##### Record types
+
+When a resource is returning a record type, each field of the record can be queried separately.
+Each record type is mapped to a GraphQL `OBJECT` type, and the field of the record type are mapped to the fields of the `OBJECT` type.
+
+```ballerina
+public type Person record {|
+    string name;
+    int age;
+|};
+
+resource function get profile() returns Person {
+    return { name: "Walter White", age: 51 };
+}
+```
+
+This will generate the following schema
+```
+type Query {
+    profile: Person!
+}
+
+type Person {
+    name: String!
+    age: Int!
+}
+```
+
+This can be queried using the following document:
+```
+{
+    profile {
+        name
+        age
+    }
+}
+```
+
+Result:
+```json
+{
+    "data": {
+        "profile": {
+            "name": "Walter White",
+            "age": 51
+        }
+    }
+}
+```
+
+Each field can be queried separately as shown in the following document:
+```
+{
+    profile {
+        name
+    }
+}
+```
+
+Result:
+```json
+{
+    "data": {
+        "profile": {
+            "name": "Walter White"
+        }
+    }
+}
+```
+
+#### Service Objects
+When a resource function returns a service object, the Service type is mapped to a GraphQL `OBJECT` type, and the resource functions of the service type will be mapped as the fields of the `OBJECT`.
+
+When a service type is returned from a `graphql:Service`, the returning service type should also follow the rules of the `graphql:Service`, explained above.
+
+```ballerina
+import ballerina/graphql;
+
+service graphql:Service /graphql on new graphql:Listener(4000) {
+    resource function get profile() returns Person {
+        return new("Walter White", 51);
+    }
+}
+
+service class Person {
+    private string name;
+    private int age;
+    
+    public function init(string name, int age) {
+        self.name = name;
+        self.age = age;
+    }
+    
+    resource function get name() returns string {
+        return self.name;
+    }
+    
+    resource function get age() returns int {
+        return self.age;
+    }
+}
+```
+
+This will generate the following schema:
+
+```
+type Query {
+    profile: Person!
+}
+
+type Person {
+    name: String!
+    age: Int!
+}
+```
+
+This can be queried using the following document:
```

Comment:
```suggestion
This can be queried using the following document:
```
Is it correct to say "the following document"? Or, should it be "the following code"? Please check all occurrences.

Reply:
Same as above